src/services/DocumentSelectionRequestProcessor.ts

```diff
@@ -101,11 +101,17 @@ export class DocumentSelectionRequestProcessor {
 
   		try {
   			yotiSessionId = await this.createSessionGenerateInstructions(personDetails, f2fSessionInfo, postOfficeSelection, selectedDocument, countryCode);
-  			await this.postToGovNotify(f2fSessionInfo.sessionId, yotiSessionId, personDetails);
-  			await this.f2fService.updateSessionWithYotiIdAndStatus(f2fSessionInfo.sessionId, yotiSessionId, AuthSessionState.F2F_YOTI_SESSION_CREATED);
-  			const updatedTtl = absoluteTimeNow() + this.environmentVariables.authSessionTtlInSecs();
-  			await this.f2fService.updateSessionTtl(f2fSessionInfo.sessionId, updatedTtl, this.environmentVariables.sessionTable());
-  			await this.f2fService.updateSessionTtl(f2fSessionInfo.sessionId, updatedTtl, this.environmentVariables.personIdentityTableName());
+			  if (yotiSessionId) {
+				  await this.postToGovNotify(f2fSessionInfo.sessionId, yotiSessionId, personDetails);
+				  await this.f2fService.updateSessionWithYotiIdAndStatus(f2fSessionInfo.sessionId, yotiSessionId, AuthSessionState.F2F_YOTI_SESSION_CREATED);
+				  const updatedTtl = absoluteTimeNow() + this.environmentVariables.authSessionTtlInSecs();
+				  await this.f2fService.updateSessionTtl(f2fSessionInfo.sessionId, updatedTtl, this.environmentVariables.sessionTable());
+				  await this.f2fService.updateSessionTtl(f2fSessionInfo.sessionId, updatedTtl, this.environmentVariables.personIdentityTableName());
+			  } else {
+				  this.logger.error(`No session foud with yotiSessionId ${yotiSessionId}`);
+				  throw new AppError(HttpCodesEnum.BAD_REQUEST, `No session foud with yotiSessionId ${yotiSessionId}`);
```

Comment:
nit: found in 111 and 112

Reply:
Should it not log and throw the error then?